api/institutions/authentication.py

```diff
@@ -386,4 +388,13 @@ def authenticate(self, request):
                 sso_department=department,
             )
 
+        # Storage region is only updated if the user is created via institutional SSO; the region will be set to the
+        # institution's preferred one if the user's current region is not in the institution's default region list.
+        if is_created:
+            user_settings = OSFStorageUserSettings.objects.get(owner=user)
+            institution_region_list = institution.storage_regions.all()
+            if institution_region_list and user_settings.default_region not in institution_region_list:
+                user_settings.default_region = institution_region_list.get(institutionstorageregion__is_preferred=True)
+                user_settings.save()
```

Comment:
For final CR: this part has been re-written (simplified) with newly added constraint